### PR TITLE
demo(table): using TenStack with editable cells

### PR DESCRIPTION
**Pull request by Lunkan89**
Example of editable cells with `TdsTable` using TanStack react table

### CODE REVIEW

**Review by nathalielindqvist — COMMENTED** [review]
When clicking directly on the text in the cells, it's possible to "select" several cells and change their styling. They also shift place in the Table itself when clicking on adjacent cells, see screen recording:

https://github.com/scania-digital-design-system/tegel-react-demo/assets/48509966/cfb8ee6b-3b4b-4593-9218-b2e194aae540


Also, I'm not sure if this is in the scope for this PR, but the tabbing is a bit strange in the Table. It skips several cells after a while and doesn't "open" the cells to be editable, see screen recording:

https://github.com/scania-digital-design-system/tegel-react-demo/assets/48509966/61e1ccc3-5751-4da2-953c-d4b5531d634d

File: src/pages/TanstackPage/EditableCell.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  return !isFocused ? (
+    <TdsBodyCell cellKey={id}>
+      <input
+        style={{ border: 'none', backgroundColor: 'transparent' }}
+        defaultValue={value as string}
+        onFocus={() => setIsFocused(true)}
+      />
+    </TdsBodyCell>
+  ) : (
+    <TdsTextField
```
[the remark this answers is not in the document]
Comment (Bickson) [reply src/pages/TanstackPage/EditableCell.tsx:65]: `TdsTextField` is not nested in the `TdsBodyCell`, using it to get simular spacing. Can remove it and style the input instead.

I'll take a look at the rest :)
Thanks!

[the remark this answers is not in the document]
Comment (Lunkan89) [reply src/pages/TanstackPage/EditableCell.tsx:65]: `TdsTextField` is not nested in the `TdsBodyCell`, using it to get simular spacing. Can remove it and style the input instead.

I'll take a look at the rest :)
Thanks!

**Comment by theJohnnyMe** [comment]
@Lunkan89 nice work so far!
 I did push a couple of things:
 - Added link in side menu navigation so anyone can easily access the page
 - Update the code a bit so the cell has a cellKey that has the same value as the header of the same column. I did it to re-enable the hover status on the column level.
 
 As for Natahlie's comment, yes, we would need to provide styling for it. Consider looking into tds-text-field component and see which values we use in _vars file. As Tegel is installed here, same CSS variables are available here too.

**Comment by Lunkan89** [comment]
@theJohnnyMe & @nathalielindqvist  not using `tds-text-field`

**Comment by theJohnnyMe** [comment]
Nice work Johan! I think this is in a good state for presentation purposes.
It is a similar situation as in column search and tds-text-field, lets discuss it with Laurens once he is back.
There is a small thing regarding the hover and color change I would like to show you tomorrow, so lets discuss it tomorrow morning :)

File: src/pages/TanstackPage/EditableCell.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  return (
+    <div style={{ display: 'table-cell' }} key={id} tabIndex={index}>
+      <input
+        ref={inputRef}
+        style={{}}
+        value={value}
+        onChange={(e: any) => setValue(e.target.value)}
+        onBlur={(e: any) => {
+          table.options.meta?.updateData(index, id, e.target.value);
```
Comment (Lunkan89) [inline src/pages/TanstackPage/EditableCell.tsx:52]: this changes the value in the tanstack table object

**Pull request closed without merging**